hare/mock_oracle.go

```diff
@@ -28,6 +29,40 @@ type Rolacle interface {
 	Eligible(instanceID uint32, committeeSize int, pubKey string, proof []byte) bool
 }
 
+type HareRolacle interface {
+	Eligible(instanceId InstanceId, k int32, pubKey string, proof []byte) bool
+}
+
+type hareRolacle struct {
+	oracle        Rolacle
+	committeeSize int
+}
+
+func newHareOracle(oracle Rolacle, committeeSize int) *hareRolacle {
+	return &hareRolacle{oracle, committeeSize}
+}
+
+func (hr *hareRolacle) Eligible(instanceId InstanceId, k int32, pubKey string, proof []byte) bool {
+	return hr.oracle.Eligible(hashInstanceAndK(instanceId, k), expectedCommitteeSize(k, hr.committeeSize), pubKey, proof)
+}
+
+func hashInstanceAndK(instanceID InstanceId, K int32) uint32 {
+	kInBytes := make([]byte, 4)
+	binary.LittleEndian.PutUint32(kInBytes, uint32(K))
+	h := newHasherU32()
+	val := h.Hash(instanceID.Bytes(), kInBytes)
+	return val
+}
+
+func expectedCommitteeSize(k int32, n int) int {
+	if k%4 == Round2 {
```

Comment:
please comment to explain why this is happening

Reply:
Added